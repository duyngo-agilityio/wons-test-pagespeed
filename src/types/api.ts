export enum Method {
  Get = 'GET',
  Post = 'POST',
  Put = 'PUT',
  Patch = 'PATCH',
  Delete = 'DELETE',
}
