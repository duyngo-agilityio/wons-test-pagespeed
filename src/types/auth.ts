// Models
import { User } from '@/models';

export interface SignInFormData {
  identifier: string;
  password: string;
}

export interface AuthResponse {
  jwt: string;
  user: User | null;
  error: string;
}

export type TSignUpPayload = {
  fullName: string;
  username: string;
  email: string;
  password: string;
};

export type TSignUpResponse = { jwt: string; user: Omit<User, 'password'> };
