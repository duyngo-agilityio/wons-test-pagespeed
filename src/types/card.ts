export enum Level {
  Low = 'low',
  Medium = 'medium',
  High = 'high',
}
