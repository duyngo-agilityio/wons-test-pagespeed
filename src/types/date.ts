import { DateValue } from '@internationalized/date';

export interface DateRangeState {
  start: DateValue;
  end: DateValue;
}
