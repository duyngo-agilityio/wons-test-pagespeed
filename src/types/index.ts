export * from './auth';
export * from './date';
export * from './params';
export * from './select';
export * from './strapi';
export * from './table';
export * from './toast';
export * from './invoice';
export * from './customer';
