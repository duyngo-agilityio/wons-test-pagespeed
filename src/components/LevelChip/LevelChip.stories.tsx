import type { Meta, StoryObj } from '@storybook/react';

// Components
import LevelChip from './index';

// Types
import { Level } from '@/types';

const meta: Meta<typeof LevelChip> = {
  title: 'Components/LevelChip',
  component: LevelChip,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
};

export default meta;
type Story = StoryObj<typeof LevelChip>;

export const Low: Story = {
  args: {
    level: Level.Low,
    size: 'md',
  },
};

export const Medium: Story = {
  args: {
    level: Level.Medium,
    size: 'md',
  },
};

export const High: Story = {
  args: {
    level: Level.High,
    size: 'md',
  },
};
