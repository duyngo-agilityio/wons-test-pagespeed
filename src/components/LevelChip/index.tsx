import { memo } from 'react';

// Libs
import { Chip } from '@nextui-org/react';

// Types
import { TSize, Level } from '@/types';

type TLevelCardProps = {
  level: Level;
  size?: TSize;
};

const LevelChip = ({ level, size = 'md' }: TLevelCardProps): JSX.Element => {
  const chipClasses = {
    high: {
      bg: 'bg-gray-200',
      color: 'text-white dark:text-gray-600',
    },
    medium: {
      bg: 'bg-blue-500',
      color: 'text-white dark:text-gray-600',
    },
    low: {
      bg: 'bg-pink-500',
      color: 'text-white dark:text-gray-600',
    },
  };

  const defaultChipSize = 'px-[22px] h-[30px] max-h-[30px] rounded-[22px]';

  const chipSize: Partial<Record<TSize, string | undefined>> = {
    md: defaultChipSize,
    sm: 'px-2 rounded-[20px]',
  };

  const chipClass = chipSize[size] ?? defaultChipSize;

  return (
    <Chip
      classNames={{
        base: [`${chipClass}`, `${chipClasses[level]?.bg}`],
        content: [
          'text-center text-md capitalize',
          `${chipClasses[level]?.color}`,
        ],
      }}
    >
      {level}
    </Chip>
  );
};

export default memo(LevelChip);
