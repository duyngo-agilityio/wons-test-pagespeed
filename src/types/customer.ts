// Models
import { ICustomer } from '@/models';

// Types
import { StrapiModel, StrapiResponse } from '@/types';

export type TCustomerDataResponse = StrapiModel<ICustomer>;

export type TCustomerListResponse = StrapiResponse<TCustomerDataResponse[]>;
