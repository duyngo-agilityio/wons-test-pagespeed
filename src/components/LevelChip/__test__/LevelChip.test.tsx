import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import LevelChip from '../index';

// Types
import { Level } from '@/types';

describe('LevelChip Component', () => {
  it('matches snapshot when level is medium and size is md', () => {
    const { container } = render(<LevelChip level={Level.Medium} size="md" />);
    expect(container).toMatchSnapshot();
  });

  it('applies default chip size when size is undefined', () => {
    const { container } = render(<LevelChip level={Level.Medium} />);

    expect(container.firstChild).toHaveClass(
      'px-[22px] h-[30px] max-h-[30px] rounded-[22px]',
    );
  });

  it('applies default chip size when size is invalid', () => {
    const { container } = render(
      <LevelChip
        level={Level.Medium}
        size={'invalid-size' as unknown as 'md' | 'lg' | 'sm'}
      />,
    );

    expect(container.firstChild).toHaveClass(
      'px-[22px] h-[30px] max-h-[30px] rounded-[22px]',
    );
  });
});
