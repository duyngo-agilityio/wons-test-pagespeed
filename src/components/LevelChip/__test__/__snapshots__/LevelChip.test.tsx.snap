// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelChip Component matches snapshot when level is medium and size is md 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap text-small text-default-foreground px-[22px] h-[30px] max-h-[30px] rounded-[22px] bg-blue-500 dark:bg-blue-200"
  >
    <span
      class="flex-1 font-normal px-2 text-center text-md capitalize text-white dark:text-gray-600"
    >
      medium
    </span>
  </div>
</div>
`;
